src/main/java/sirius/db/es/ElasticQuery.java

```diff
@@ -1465,18 +1465,31 @@ public void truncate() {
      * <p>
      * Use this for larger result sets where integrity and constraints do not matter or are managed manually.
      *
-     * @throws OptimisticLockException if one of the documents was modified during the runtime of the truncate
+     * @throws OptimisticLockException               if one of the documents was modified during the runtime of the truncate
+     * @throws sirius.kernel.health.HandledException if the {@linkplain LowLevelClient#deleteByQuery(String, String, JSONObject) deleteByQuery} request aborted due to any unrecoverable errors during the process
      */
     public void tryTruncate() throws OptimisticLockException {
         if (forceFail) {
             return;
         }
 
         String filteredRouting = checkRouting(Elastic.RoutingAccessMode.WRITE);
-        elastic.getLowLevelClient()
-               .deleteByQuery(computeEffectiveIndexName(elastic::determineWriteAlias),
-                              filteredRouting,
-                              buildSimplePayload());
+        JSONObject deleteByQueryResponse = elastic.getLowLevelClient()
+                                                  .deleteByQuery(computeEffectiveIndexName(elastic::determineWriteAlias),
+                                                                 filteredRouting,
+                                                                 buildSimplePayload());
+        if (Boolean.TRUE.equals(deleteByQueryResponse.getBoolean("timed_out"))
+            || deleteByQueryResponse.getIntValue("version_conflicts") > 0) {
+            throw new OptimisticLockException("Truncate timed out or had version conflicts:\n"
+                                              + deleteByQueryResponse.toJSONString());
+        }
+        if (deleteByQueryResponse.getJSONArray("failures") != null && !deleteByQueryResponse.getJSONArray("failures")
+                                                                                            .isEmpty()) {
+            throw Exceptions.createHandled()
+                            .withSystemErrorMessage("Truncate aborted due to unrecoverable error(s):\n"
+                                                    + deleteByQueryResponse.toJSONString())
```

Comment:
no we don't want to reveal this to the user.

Log to syslog and create a handled exception with a short error like "Truncate aborted due to unrecoverable error(s)."